crates/milli/src/search/facet/filter.rs

```diff
@@ -434,36 +466,56 @@ impl<'a> Filter<'a> {
                 }
             }
             FilterCondition::In { fid, els } => {
-                if crate::is_faceted(fid.value(), filterable_fields) {
-                    let field_ids_map = index.fields_ids_map(rtxn)?;
-                    if let Some(fid) = field_ids_map.id(fid.value()) {
-                        els.iter()
-                            .map(|el| Condition::Equal(el.clone()))
-                            .map(|op| Self::evaluate_operator(rtxn, index, fid, universe, &op))
-                            .union()
-                    } else {
-                        Ok(RoaringBitmap::new())
+                match matching_features(fid.value(), filterable_fields) {
+                    Some(features) if features.is_filterable() => {
+                        let field_ids_map = index.fields_ids_map(rtxn)?;
+                        if let Some(fid) = field_ids_map.id(fid.value()) {
+                            els.iter()
+                                .map(|el| Condition::Equal(el.clone()))
+                                .map(|op| {
+                                    Self::evaluate_operator(
+                                        rtxn, index, fid, universe, &op, &features,
+                                    )
+                                })
+                                .union()
+                        } else {
+                            Ok(RoaringBitmap::new())
+                        }
+                    }
+                    _ => {
+                        let field_ids_map = index.fields_ids_map(rtxn)?;
+                        Err(fid.as_external_error(FilterError::AttributeNotFilterable {
+                            attribute: fid.value(),
+                            filterable_fields: filtered_matching_field_names(
+                                filterable_fields,
+                                &field_ids_map,
+                                &|features| features.is_filterable(),
+                            ),
+                        }))?
                     }
-                } else {
-                    Err(fid.as_external_error(FilterError::AttributeNotFilterable {
-                        attribute: fid.value(),
-                        filterable_fields: filterable_fields.clone(),
-                    }))?
                 }
             }
             FilterCondition::Condition { fid, op } => {
-                if crate::is_faceted(fid.value(), filterable_fields) {
-                    let field_ids_map = index.fields_ids_map(rtxn)?;
-                    if let Some(fid) = field_ids_map.id(fid.value()) {
-                        Self::evaluate_operator(rtxn, index, fid, universe, op)
-                    } else {
-                        Ok(RoaringBitmap::new())
+                match matching_features(fid.value(), filterable_fields) {
+                    Some(features) if features.is_filterable() => {
+                        let field_ids_map = index.fields_ids_map(rtxn)?;
```

Comment:
I don't think this is new from this PR, but I also don't think we should be deserializing a whole new `field_ids_map` on each condition of the filter...

I think it should be spawned in `evaluate` and passed to all subfunctions instead.